lib/usagereporter/teleport/types.go

```diff
@@ -407,26 +410,43 @@ func (u *ResourceHeartbeatEvent) Anonymize(a utils.Anonymizer) prehogv1.SubmitEv
 	}
 }
 
+type UserMetadataClient interface {
+	GetUsername(context.Context) (string, error)
+	GetUserIsSSO(context.Context, string) (bool, error)
+}
+
 // ConvertUsageEvent converts a usage event from an API object into an
 // anonymizable event. All events that can be submitted externally via the Auth
 // API need to be defined here.
-func ConvertUsageEvent(event *usageeventsv1.UsageEventOneOf, identityUsername string) (Anonymizable, error) {
+func ConvertUsageEvent(ctx context.Context, event *usageeventsv1.UsageEventOneOf, clt UserMetadataClient) (Anonymizable, error) {
```

Comment:
Absolutely :+1: 
One of the reasons that I added this client is to prevent unnecessary network calls.
If we could include the IsSSO user in the identity, then it would be even faster and maybe we could pass the values directly and remove the client altogether

I'll change things and ping you for another pass :+1: